strawberryfields/gbs/subgraph.py

```diff
@@ -183,193 +178,98 @@ def random_search(
     """
-    resize_options = {**RESIZE_DEFAULTS, **(resize_options or {})}
+    subgraph = set(subgraph)
+    if not subgraph.issubset(graph.nodes):
+        raise ValueError("Input is not a valid subgraph")
 
-    try:
-        target = int(target)
-    except TypeError:
-        raise TypeError("target must be an integer")
+    nodes = graph.nodes()
+    all_sizes = set(range(1, len(nodes)))
 
-    if not 2 < target < graph.number_of_nodes():
-        raise ValueError(
-            "target must be greater than two and less than the number of nodes in "
-            "the input graph"
-        )
+    if sizes is None:
+        sizes = all_sizes
+    else:
+        sizes = set(sizes)
 
-    method = resize_options["method"]
+        if not sizes.issubset(all_sizes):
+            raise ValueError("Requested sizes must be within size range of graph")
 
-    if not callable(method):
-        method = RESIZE_DICT[method]
+    starting_size = len(subgraph)
+    max_size = max(sizes)
+    min_size = min(sizes)
 
-    return method(subgraphs=subgraphs, graph=graph, target=target, resize_options=resize_options)
+    if starting_size in sizes:
+        resized = {starting_size: sorted(subgraph)}
+    else:
+        resized = {}
 
+    if max_size > starting_size:
 
-def greedy_density(
-        subgraphs: Iterable, graph: nx.Graph, target: int, resize_options: Optional[dict] = None
-) -> list:
-    """Method to greedily resize subgraphs based upon density.
+        grow_subgraph = graph.subgraph(subgraph).copy()
 
-    This function uses a greedy approach to iteratively add/remove nodes to an input subgraph.
-    Suppose the input subgraph has :math:`M` nodes. When shrinking, the algorithm considers
-    all :math:`M-1` node subgraphs of the input subgraph, and picks the one with the greatest
-    density. It then considers all :math:`M-2` node subgraphs of the chosen :math:`M-1` node
-    subgraph, and continues until the desired subgraph size has been reached.
+        while grow_subgraph.order() < max_size:
+            grow_nodes = grow_subgraph.nodes()
+            complement_nodes = nodes - grow_nodes
 
-    When growing, the algorithm considers all :math:`M+1` node subgraphs of the input graph given by
-    combining the input :math:`M` node subgraph with another node from the remainder of the input
-    graph, and picks the one with the greatest density. It then considers all :math:`M+2` node
-    subgraphs given by combining the chosen :math:`M+1` node subgraph with nodes remaining from
-    the overall graph, and continues until the desired subgraph size has been reached.
+            degrees = [
+                (c, graph.subgraph(list(grow_nodes) + [c]).degree()[c]) for c in complement_nodes
+            ]
+            np.random.shuffle(degrees)
 
-    ``resize_options`` is not currently in use in :func:`greedy_density`.
+            to_add = max(degrees, key=lambda x: x[1])
```

Comment:
Is there no random breaking of ties as we do in `clique_shrink`?  Or is this done implicitly by `random.shuffle`?